source/GStreamerMSEMediaPlayerClient.cpp

```diff
@@ -444,6 +458,70 @@ double GStreamerMSEMediaPlayerClient::getVolume()
     return volume;
 }
 
+void GStreamerMSEMediaPlayerClient::setAudioStreamsInfo(int32_t audioStreams, bool isAudioOnly)
+{
+    mBackendQueue.callInEventLoop(
+        [&]()
+        {
+            if (mAudioStreams == UNKNOWN_STREAMS_NUMBER)
+            {
+                mAudioStreams = audioStreams;
+                GST_INFO("Set audio streams number to %d", mAudioStreams);
+            }
+
+            if (mVideoStreams == UNKNOWN_STREAMS_NUMBER && isAudioOnly)
+            {
+                mVideoStreams = 0;
+                GST_INFO("Set video only session");
+            }
+        });
+}
+
+void GStreamerMSEMediaPlayerClient::setVideoStreamsInfo(int32_t videoStreams, bool isVideoOnly)
+{
+    mBackendQueue.callInEventLoop(
+        [&]()
+        {
+            if (mVideoStreams == UNKNOWN_STREAMS_NUMBER)
+            {
+                mVideoStreams = videoStreams;
+                GST_INFO("Set video streams number to %d", mVideoStreams);
+            }
+
+            if (mAudioStreams == UNKNOWN_STREAMS_NUMBER && isVideoOnly)
+            {
+                mAudioStreams = 0;
+                GST_INFO("Set audio only session");
+            }
+        });
+}
+
+bool GStreamerMSEMediaPlayerClient::areAllStreamsAttached()
```

Comment:
Should this be a private method and not posted on the backend queue? We only call this locally and already in an event loop.